services/rssbot/rssbot.go

```diff
@@ -415,7 +418,11 @@ func (s *Service) sendToRooms(cli types.MatrixClient, feedURL string, feed *gofe
 	logger.Info("Sending new feed item")
 	for _, roomID := range s.Feeds[feedURL].Rooms {
 		if _, err := cli.SendMessageEvent(roomID, mevt.EventMessage, itemToHTML(feed, item)); err != nil {
-			logger.WithError(err).WithField("room_id", roomID).Error("Failed to send to room")
+			if err.Code == 429  {
```

Comment:
I think it should be more like this, the Go `error` interface doesn't have a `.Code` field:

```suggestion
			if httpErr := err.(mautrix.HTTPError); httpErr.IsStatus(429) {
```

(may require updating mautrix-go)

Reply:
Or with the fancy modern [`errors.As()`](https://golang.org/pkg/errors/#As) function

```suggestion
			if httpErr := mautrix.HTTPError{}; errors.As(err, &httpErr) && httpErr.IsStatus(429)  {
```